app.js

```diff
@@ -147,6 +168,57 @@ router.post('/account/:phoneNumber/:accountId/validateCode', async ctx => {
     ctx.body = {};
 });
 
+function recoveryMethodsFor(account) {
+    return {
+        email: account.email || null,
+        emailAddedAt: account.emailAddedAt || null,
+        phoneAddedAt: account.phoneAddedAt || null,
+        phoneNumber: account.phoneNumber || null,
+        phraseAddedAt: account.phraseAddedAt || null,
+    };
+}
+
+router.post('/account/:accountId/recoveryMethods', async ctx => {
+    const { accountId } = ctx.params;
+    const { securityCode, signature } = ctx.request.body;
+
+    const account = await models.Account.findOne({ where: { accountId } });
+
+    await verifyAccountOwnership({ ctx, accountId, securityCode, signature });
+
+    ctx.body = recoveryMethodsFor(account || {});
+});
+
+const recoveryMethods = ['phone', 'email', 'phrase'];
+
+router.post('/account/deleteRecoveryMethod', async ctx => {
+    const { accountId, recoveryMethod, securityCode, signature } = ctx.request.body;
+
+    if (!recoveryMethods.includes(recoveryMethod)) {
+        ctx.throw(400, `Given recoveryMethod '${recoveryMethod}' invalid; must be one of: ${recoveryMethods.join(', ')}`);
+    }
+
+    const account = await models.Account.findOne({ where: { accountId } });
+
+    if (!account) {
+        ctx.throw(404, `Could not find account with accountId: '${accountId}'`);
+    }
```

Comment:
Should IMO live in middleware, similar to https://github.com/nearprotocol/near-fiddle-api/blob/c0ecec0b27fba1a0c961e7e0223c961ac113d50d/app.js#L59